### DISCUSSION TITLE
Smooth Player Teleportation

### DISCUSSION THREAD

**Discussion by ghost** [Minestom/Minestom#1068] [Help] [answered]
Been poking around Minestom and have been thinking about porting my movecraft-like paper plugin to use it. However, one concern I've had is smooth player teleportation. (The player is teleported, but can still look around without their view being reset) 

Related [Spigot Resource](https://www.spigotmc.org/threads/teleport-player-smoothly.317416/)

The code I'm currently using to teleport the player is derived from [this](https://github.com/HorizonsEndMC/IonCore/blob/main/src/main/kotlin/net/starlegacy/util/ConnectionUtils.kt ) which uses `net.minecraft.network.protocol.game.ClientboundPlayerPositionPacket`. Is this equivalent to `net.minestom.server.network.packet.client.play.ClientPlayerPositionPacket`?

Sorry if this is a dumb question, I've never really used Minestom before. The whole effort was to bypass the stupidity of Bukkit's built-in teleport, but I don't know if Minestom even has the same issue in the first place.

At the end of the day, **all I really need is a method of teleporting the player without resetting their view**, and want to make sure it's possible before I invest time in trying to use Minestom.

**Comment by ryleu** [comment]
i believe you can just use a `Point` rather than a `Position` and that will work. 

bear in mind that Minestom IS NOT LIKE BUKKIT. you will be programming much of the game's functionality yourself, so it's important to know that going into it.

**Reply by ghost** [reply]
> bear in mind that Minestom IS NOT LIKE BUKKIT. you will be programming much of the game's functionality yourself, so it's important to know that going into it.

Yeah I'm aware. That's part of the reason I want to try it out, as I keep finding myself wanting to remove vanilla functionality to replace it.

**Comment by GreatWyrm** [comment, marked as answer]
The ClientboundPlayerPositionPacket has its equivalent in Minestom as `net.minestom.server.network.packet.server.play.PlayerPositionAndLookPacket`.

Packets in the packet.client package are received from the client, and the packets in the packet.server are sent to the client (and are therefore clientbound). Hope that explanation helps!

**Reply by ghost** [reply]
Thanks!
